src/helper/proxy/index.ts

```diff
@@ -29,14 +29,37 @@ interface ProxyFetch {
   (input: string | URL | Request, init?: ProxyRequestInit): Promise<Response>
 }
 
+const propagateRequestHeaders = (headers: Headers, init: ProxyRequestInit | undefined) => {
+  let requestHeaders
+
+  if (Array.isArray(init?.headers)) {
+    requestHeaders = init.headers
+  } else if (init?.headers) {
+    requestHeaders = Object.entries(init.headers)
+  }
+
+  if (requestHeaders) {
+    for (const [name, value] of requestHeaders) {
+      if (value === undefined) {
+        headers.delete(name)
+      } else {
+        headers.append(name, value)
+      }
```

Comment:
Unsure if this is right..

It's deleting a header when the value is `undefined`, otherwise the value is appended. This means values in both the `ProxyRequestInit` and the request will be merged, rather than the value from the request overwriting the value from `ProxyRequestInit`, for example.